metamist/parser/generic_parser.py

```diff
@@ -662,6 +683,31 @@ def prepare_summary(
 
         return summary
 
+    def prepare_detail(self, samples: list[ParsedSample]):
+        """Uses tabulate to print a detailed summary of the samples being inserted / updated"""
+        sample_participants = {}
+        for sample in samples:
+            sample_participants[sample.external_sid] = sample.participant.external_pid if sample.participant else None
+        sample_sequencing_groups = {sample.external_sid: sample.sequencing_groups for sample in samples}
+
+        details = []
+        for sample, participant in sample_participants.items():
+            for sg in sample_sequencing_groups[sample]:
+                sg_details = {
+                    'Participant': '',
```

Comment:
What about:
'Participant': participant if participant else ''
Then you can remove that if condition afterwards.